metecho/api/jobs.py

```diff
@@ -447,6 +447,7 @@ def create_pr(
 def delete_scratch_org(scratch_org, *, originating_user_id):
     try:
         delete_org(scratch_org)
+        scratch_org.refresh_from_db()
```

Comment:
@wlonk So when a Task with an Org is deleted, here's the flow:

- SoftDeleteMixin for task runs, finds child org
- SoftDeleteQuerySet for org runs, queuing job
- job runs, deleting org on SF and then triggering org.delete()
- SoftDeleteMixin for org runs again

Without this line, the second run of the SoftDeleteMixin for the org still found `self.deleted_at is None`, which caused _another_ delete_org job to be queued.